src/lib.rs

```diff
@@ -81,16 +81,49 @@ fn new_impl(ast: &syn::MacroInput, fields: Option<&[syn::Field]>,
             format!("Constructs a new `{}::{}`.", name, variant),
         ),
     };
+    let attrs = collect_parent_attrs(&ast.attrs);
+    let attrs = quote![#(#attrs),*];
     quote! {
         impl #impl_generics #name #ty_generics #where_clause {
             #[doc = #doc]
+            #attrs
             pub fn #new(#(#args),*) -> Self {
                 #name #qual #inits
             }
         }
     }
 }
 
+fn collect_parent_attrs(attrs: &[syn::Attribute]) -> Vec<syn::Attribute> {
```

Comment:
Either in the function name or a comment, you should make clear that this collects only attrs which are lints, not all attrs.